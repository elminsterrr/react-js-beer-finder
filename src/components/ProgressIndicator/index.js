import React from 'react';
import CircularProgress from 'material-ui/CircularProgress';

// import './style.css';

const ProgressIndicator = () => (
  <div className="ProgressIndicator-div">
    Searching...
    <CircularProgress size={60} thickness={7} />
  </div>
);

export default ProgressIndicator;
