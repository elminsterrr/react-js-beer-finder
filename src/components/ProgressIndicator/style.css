.ProgressIndicator-container {
  width: 15%;
  margin: 25px auto 0 auto;
  text-align: center;
}

.ProgressIndicator-circular-progress {
  margin: 25px 0;
}
