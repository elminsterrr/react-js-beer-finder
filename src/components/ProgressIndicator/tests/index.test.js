import React from 'react';
import { shallow } from 'enzyme';
import ProgressIndicator from '../';

describe('ProgressIndicator', () => {
  const progressInd = shallow(<ProgressIndicator />);

  it('renders correctly', () => {
    expect(progressInd).toMatchSnapshot();
  });

  it('renders `CircularProgress` indicator', () => {
    expect(
      progressInd.find('.ProgressIndicator-circular-progress').children().length
    ).toEqual(1);
  });
});
