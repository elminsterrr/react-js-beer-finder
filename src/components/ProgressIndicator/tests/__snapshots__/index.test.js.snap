// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProgressIndicator renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ProgressIndicator />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <WithStyles(CircularProgress)
        style={
                Object {
                        "color": "#fbc02d",
                      }
        }
        thickness={7}
/>,
      "className": "ProgressIndicator-container",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "style": Object {
          "color": "#fbc02d",
        },
        "thickness": 7,
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <WithStyles(CircularProgress)
          style={
                    Object {
                              "color": "#fbc02d",
                            }
          }
          thickness={7}
/>,
        "className": "ProgressIndicator-container",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "style": Object {
            "color": "#fbc02d",
          },
          "thickness": 7,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
